### PR TITLE
[CHK-465] Add payment-gateway-service client

### PR DESCRIPTION
**Pull request by giovanniberti**
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: api-spec/api.yaml [hunk 1]
```diff
@@ -79,6 +79,18 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/ProblemJson"
+        409:
+          description: Transaction already authorized
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemJson"
+        504:
```
[the remark this answers is not in the document]
Comment (giovanniberti) [reply api-spec/api.yaml:88]: For documentation purposes? Or is there any scenario where we can answer with a 500 and a meaningful response?

[the remark this answers is not in the document]
Comment (infantesimone) [reply api-spec/api.yaml:88]: only for docs

File: src/main/java/it/pagopa/transactions/exceptions/UnsatisfiablePspRequestException.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package it.pagopa.transactions.exceptions;
+
+import it.pagopa.generated.transactions.server.model.RequestAuthorizationRequestDto;
+import it.pagopa.transactions.domain.PaymentToken;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(value = HttpStatus.UNPROCESSABLE_ENTITY)
```
Comment (infantesimone) [inline src/main/java/it/pagopa/transactions/exceptions/UnsatisfiablePspRequestException.java:8]: 422?

Comment (giovanniberti) [reply src/main/java/it/pagopa/transactions/exceptions/UnsatisfiablePspRequestException.java:8]: Forgot to change :) ty

**Review by infantesimone — APPROVED** [review]

**Pull request merged**